DebugGUI/src/DebugGUI.cxx

```diff
@@ -54,31 +55,98 @@ void* initGUI(const char* name, void(*error_callback)(int, char const*descriptio
   return window;
 }
 
-/// @return true if we do not need to exit, false if we do.
-bool pollGUI(void* context, std::function<void(void)> guiCallback)
+// fills a stream with drawing data in JSON format
+// the JSON is composed of a list of draw commands
+/// FIXME: document the actual schema of the format.
+void getFrameJSON(void *data, std::ostream& json_data)
+{
+  auto draw_data = (ImDrawData*)data;
+
+  json_data << "[";
+
+  for (int cmd_id = 0; cmd_id < draw_data->CmdListsCount; ++cmd_id) {
+    const auto cmd_list = draw_data->CmdLists[cmd_id];
+    const auto vtx_buffer = cmd_list->VtxBuffer;
+    const auto idx_buffer = cmd_list->IdxBuffer;
+    const auto cmd_buffer = cmd_list->CmdBuffer;
+
+    json_data << "{\"vtx\":[";
+    for (int i = 0; i < vtx_buffer.size(); ++i) {
+      auto v = vtx_buffer[i];
+      json_data << "[" << v.pos.x << "," << v.pos.y << "," << v.col << ',' << v.uv.x << "," << v.uv.y << "]";
+      if (i < vtx_buffer.size() - 1) json_data << ",";
+    }
+
+    json_data << "],\"idx\":[";
+    for (int i = 0; i < idx_buffer.size(); ++i) {
+      auto id = idx_buffer[i];
+      json_data << id;
+      if (i < idx_buffer.size() - 1) json_data << ",";
+    }
+
+    json_data << "],\"cmd\":[";
+    for (int i = 0; i < cmd_buffer.size(); ++i) {
+      auto cmd = cmd_buffer[i];
+      json_data << "{\"cnt\":" << cmd.ElemCount << ", \"clp\":[" << cmd.ClipRect.x << "," << cmd.ClipRect.y << "," << cmd.ClipRect.z << "," << cmd.ClipRect.w << "]}";
+      if (i < cmd_buffer.size() - 1) json_data << ",";
+    }
+    json_data << "]}";
+    if (cmd_id < draw_data->CmdListsCount - 1) json_data << ",";
+  }
+
+  json_data << "]";
+}
+
+bool pollGUI_gl_init(void* context)
```

Comment:
Notice that as a convention in general we use `camelCase` unless the code comes from a third party, so here it should be:

```suggestion
bool pollGUIPreRender(void* context)
```

Also avoid referring to the graphics API as you will need to replicate this also in the Metal backend.